pkg/s3/s3client.go

```diff
@@ -0,0 +1,64 @@
+package s3
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+type S3Client interface {
+	DownloadObject(output io.WriterAt, bucket string, key string) (int64, error)
+	ListObjectsNonRecursive(bucket string) ([]*string, error)
+}
+
+type s3Client struct {
+	client     *s3.S3
+	downloader *s3manager.Downloader
+}
+
+func NewS3Client(sess *session.Session) S3Client {
+	return &s3Client{
+		client:     s3.New(sess),
+		downloader: s3manager.NewDownloader(sess),
+	}
+}
+
+// DownloadObject downloads a file from S3 based on the input: bucket and key, and write to the output.
+// The output can be of os.File type or WriteAtBuffer type, based on where the downloaded file should be kept
+func (s *s3Client) DownloadObject(output io.WriterAt, bucket string, key string) (int64, error) {
```

Comment:
you may consider parallelizing the download task later. Is it a blocking function call?

Reply:
Yes it's blocking. This is only used for loading the content to memory at service startup. If it fails, the service should fail. So it needs to be blocking.

Reply:
The download and caching can be a complicated process. Please add a github issue for the next iteration.

Reply:
Created #62